numpy/random/mt19937.pyx

```diff
@@ -128,121 +131,74 @@ cdef class MT19937:
 
     """
     cdef mt19937_state rng_state
-    cdef bitgen_t _bitgen
-    cdef public object capsule
-    cdef object _ctypes
-    cdef object _cffi
-    cdef public object lock
 
     def __init__(self, seed=None):
-        self.seed(seed)
-        self.lock = Lock()
-
+        BitGenerator.__init__(self, seed)
+        if isinstance(seed, SeedSequence):
+            val = self._seed_seq.generate_state(RK_STATE_LEN, np.uint32)
+            # MSB is 1; assuring non-zero initial array
+            self.rng_state.key[0] = 0x80000000UL 
+            for i in range(1, RK_STATE_LEN):
+                self.rng_state.key[i] = val[i]
+            self.rng_state.pos = i
+        else:
+            self.legacy_seeding(seed)
```

Comment:
This is [not what I was suggesting](https://github.com/numpy/numpy/pull/13780#discussion_r295563065).

`prng = MT19937(some_seed)` should be equivalent to `prng = MT19937(SeedSequence(some_seed))`, just like all of the other PRNGs in numpy. In order to get the old seeding, we need a `MT19937.legacy_seeding()` `@classmethod`: `prng = MT19937.legacy_seeding(some_seed)`. Just refactoring `MT19937.__init__()` is not what I was talking about.

Reply:
Do you mean `legacy_seeding` is an alternative instantiation method, called only by `RandomState` and only for legacy `MT19937` creation?

Reply:
Yes.

Reply:
An alternative would be to cdef legacy_seed as method so that it was only available to Cython.  This would require a MT19937 class def in a pxd so that it could be cast when needed. The `seed` interface on `RandomState` is presumably not going away, and so there needs to be some wat to reseed an MT19937 (you could always instantize and then swap the bitgen, but that seems a bit awkward).  Is there a case for a legacy_seed that for an MT19937 instance that was directly available?  Even for testing, one could stick it in a RandomState, let it so the instantiation, and then grab the MT19937.  

This said, it might not be worth the effort to hide this interface from end users.